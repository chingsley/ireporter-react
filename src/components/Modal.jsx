/* eslint-disable react/button-has-type */
import React, { Fragment } from 'react';
import Modal from 'react-modal';
import PropTypes from 'prop-types';

const ViewModal = (props) => {
  const {
    displayModal, closeModal, image, comment,
  } = props;

  return (
    <Fragment>
      <Modal
        isOpen={displayModal}
        contentLabel="Modal Option"
        ariaHideApp={false}
        className="Modal"
        overlayClassName="Overlay"
      >
        <div className="container-modal-content">
          <div className="media-container">
            <img src={image} alt="default" className="media-container__img" />
            <p className="media-container__text">
              {comment}
            </p>
          </div>
          <button className="btn-modal cancel" onClick={closeModal}>x</button>
        </div>
      </Modal>
    </Fragment>
  );
};
ViewModal.propTypes = {
  closeModal: PropTypes.func.isRequired,
  displayModal: PropTypes.bool,
  image: PropTypes.string,
  comment: PropTypes.string,
};

ViewModal.defaultProps = {
  displayModal: false,
  image: 'https://via.placeholder.com/700x400',
  comment: 'No comments provied',
};

export default ViewModal;
