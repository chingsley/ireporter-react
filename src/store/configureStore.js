import {
  applyMiddleware, createStore, combineReducers, compose,
} from 'redux';
import thunk from 'redux-thunk';
import authReducer from '../reducers/authReducer';

export default () => {
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // eslint-disable-line
  const store = createStore(
    combineReducers({
      auth: authReducer,
    }),
    composeEnhancers(applyMiddleware(thunk)),
  );
  return store;
};
