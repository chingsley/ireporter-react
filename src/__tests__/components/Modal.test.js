import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import Modal from '../../components/Modal';

const wrap = () => {
  const props = {
    history: {
      push: jest.fn(),
      location: {
        pathname: '',
      },
    },

    authorise: jest.fn(),
    isLoggedIn: false,
    fetching: false,
    formType: 'signup',
  };

  const mockStore = configureStore([thunk]);

  const store = mockStore({
    auth: {
      isLoggedIn: true,
      user: {},
    },
    fetchStatus: {
      fetching: false,
      errMessage: '',
    },
    report: {
      report: {},
    },
  });

  const shallowWrapper = shallow(<Modal {...props} />);

  const mountWrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <Modal {...props} />
      </MemoryRouter>
    </Provider>,
  );
  return { shallowWrapper, mountWrapper };
};

describe('Test the report form', () => {
  const { mountWrapper: wrapper } = wrap();
  it('should render with the correct elements', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
