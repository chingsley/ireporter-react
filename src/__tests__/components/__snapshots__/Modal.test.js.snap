// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the report form should render with the correct elements 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "rhzs8l",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "rhzs8l",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ViewModal
        authorise={[MockFunction]}
        comment="No comments provied"
        displayModal={false}
        fetching={false}
        formType="signup"
        history={
          Object {
            "location": Object {
              "pathname": "",
            },
            "push": [MockFunction],
          }
        }
        image="https://via.placeholder.com/700x400"
        isLoggedIn={false}
      >
        <Modal
          ariaHideApp={false}
          bodyOpenClassName="ReactModal__Body--open"
          className="Modal"
          closeTimeoutMS={0}
          contentLabel="Modal Option"
          isOpen={false}
          overlayClassName="Overlay"
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          role="dialog"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
        >
          <Portal
            containerInfo={
              <div
                class="ReactModalPortal"
              />
            }
          >
            <ModalPortal
              ariaHideApp={false}
              bodyOpenClassName="ReactModal__Body--open"
              className="Modal"
              closeTimeoutMS={0}
              contentLabel="Modal Option"
              defaultStyles={
                Object {
                  "content": Object {
                    "WebkitOverflowScrolling": "touch",
                    "background": "#fff",
                    "border": "1px solid #ccc",
                    "borderRadius": "4px",
                    "bottom": "40px",
                    "left": "40px",
                    "outline": "none",
                    "overflow": "auto",
                    "padding": "20px",
                    "position": "absolute",
                    "right": "40px",
                    "top": "40px",
                  },
                  "overlay": Object {
                    "backgroundColor": "rgba(255, 255, 255, 0.75)",
                    "bottom": 0,
                    "left": 0,
                    "position": "fixed",
                    "right": 0,
                    "top": 0,
                  },
                }
              }
              isOpen={false}
              overlayClassName="Overlay"
              parentSelector={[Function]}
              portalClassName="ReactModalPortal"
              role="dialog"
              shouldCloseOnEsc={true}
              shouldCloseOnOverlayClick={true}
              shouldFocusAfterRender={true}
              shouldReturnFocusAfterClose={true}
              style={
                Object {
                  "content": Object {},
                  "overlay": Object {},
                }
              }
            />
          </Portal>
        </Modal>
      </ViewModal>
    </Router>
  </MemoryRouter>
</Provider>
`;
